core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -30,7 +30,11 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
 
   def shutdown(): Unit = {
     initiateShutdown()
-    awaitShutdown()
+    if (this.isAlive) {
```

Comment:
I think it'd be clearer if this logic was moved to `awaitShutdown`.

Reply:
Makes sense, thanks!

Reply:
@dhruvilshah3 actually I think `awaitShutdown()` needs to block regardless of the thread being alive or not. There could be cases where the user calls `awaitShutdown()` from a thread before starting the `ShutdownableThread`. The goal I had here was to just short-circuit `shutdown()` if the thread was never started.